napari/layers/utils/layer_utils.py

```diff
@@ -94,8 +94,8 @@ def calc_data_range(data, rgb=False):
             slice(-4096, None),
         ]
         reduced_data = [
-            [np.max(data[sl]) for sl in slices],
-            [np.min(data[sl]) for sl in slices],
+            [np.nanmax(data[sl]) for sl in slices],
```

Comment:
I suspect @jni has the preference for a slightly more optimal syntax for handling the nanmax/ nanmin. Maybe he can post it here. I was trying to find issue where it was previously discussed but could not

Reply:
I believe his preference is to try catching the error before falling back to using `nanmax` (on the basis that `nanmax` is a lot slower, so it's good to try and do things the fast way first, then if that doesn't work move to the slower option). Please correct me if I'm wrong, @jni

Reply:
Right, I remember that discussion. I'll do that!

Reply:
Fixed, I think. It's a bit more complicated than using `np.nanmin`, because `np.inf` is also an issue. I accounted for that too, assuming that we want to ignore inf and -inf for the purpose of calculating contrast limits.